### PR TITLE
feat(components): add notebook, cell and code editor components

### PR DESCRIPTION
**Pull request by bsahitya**
## Description

Web components added:
1. Code editor
2. Notebook
3. Notebook cell

### What's included?

1. Code editor - Creates a monaco editor instance with the options passed
2. Notebook - Resembles a python notebook. Renders notebook cells as children.
3. Notebook cell - Resembles a python notebook cell

#### Test Steps

- [x] `npm run storybook`
- [x] Navigate to Notebook, Notebook cell & Code editor components
- [x] Play around with the properties

#### General Tests for Every PR

- [x] `npm run start` still works.
- [x] `npm run lint` passes.
- [x] `npm run stylelint` passes.
- [x] `npm test` passes and code coverage is not lower.
- [x] `npm run build` still works.

#### Code editor
Screenshot 2024-07-29 at 5 03 33 PM

#### Notebook cell
Screenshot 2024-07-29 at 5 04 27 PM

#### Notebook
Screenshot 2024-07-30 at 9 48 41 AM

### CODE REVIEW

File: libs/components/src/code-editor/code-editor.scss [hunk 1]
```diff
@@ -0,0 +1,26 @@
+:host {
```
Comment (owilliams320) [inline libs/components/src/code-editor/code-editor.scss:1]: Can we prefix these variables with `cv`

File: libs/components/src/notebook-cell/notebook-cell.ts [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+          @editor-blur="${() => this.setEditorFocus(false)}"
+          .code="${this.code}"
+          .language="${this.language}"
+          .options="${this.editorOptions}"
+        ></cv-code-editor>`
+      : html`<cv-code-snippet hideHeader .language="${this.language}"
+          >${this.code}</cv-code-snippet
+        >`;
+
+    const output = html` <div class="errors"><slot name="error"></slot></div>
```
Comment (owilliams320) [inline libs/components/src/notebook-cell/notebook-cell.ts:148]: can we use the page level alerts component for errors? https://teradata.github.io/covalent/docs/components/?path=/story/components-alert--page-level&args=state:negative

**Review by owilliams320 — CHANGES_REQUESTED** [review]
Just had a few comments

**Review by owilliams320 — APPROVED** [review]

File: libs/components/src/notebook/notebook.ts [hunk 3]
```diff
@@ -0,0 +1,650 @@
...
+        <cv-icon-button
+          icon="refresh"
+          @click="${() => this.dispatchCustomCellEvent('refresh')}"
+        ></cv-icon-button>
+        <cv-icon-button
+          icon="fast_forward"
+          @click="${() => this.dispatchCustomCellEvent('refresh-run-all')}"
+        ></cv-icon-button>
+        <cv-select
+          label="Cell type"
```
[the remark this answers is not in the document]
Comment (bsahitya) [reply libs/components/src/notebook/notebook.ts]: Added an input called `labels` for the labels and tooltips to be passed from the consumer.

**Pull request closed without merging**